azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -28,6 +28,35 @@
 TOPIC_POS_INPUT_NAME = 5
 
 
+class TransportAction:
```

Comment:
>TransportAction [](start = 6, length = 15)

all actions seem to have a callback - should that be part of the TransportAction contract?

Reply:
also, it sounds like we are mapping actions to MQTT Packets types. I wonder if the naming should reflect that.

---
In reply to: [266183861](https://github.com/Azure/azure-iot-sdk-python-preview/pull/63#discussion_r266183861) [](ancestors = 266183861)

Reply:
added callback.
Also, this is not a 1:1 mapping to packet types.  This is a higher level.  MethodResponse and SendMessage both result in PUBLISH operations.

---
In reply to: [266184101](https://github.com/Azure/azure-iot-sdk-python-preview/pull/63#discussion_r266184101) [](ancestors = 266184101,266183861)